src/payment/payment_store.rs

```diff
@@ -82,6 +137,73 @@ impl_writeable_tlv_based_enum!(PaymentStatus,
 	(4, Failed) => {};
 );
 
+/// Represents the kind of a payment.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum PaymentKind {
```

Comment:
Should we name this `PaymentPurpose` for a closer mapping to LDK?

Reply:
No strong opinion. Though I wonder if "kind" is just another way of saying "type". I'm a little more inclined with using the latter, but happy to defer to you here.